laws/src/main/scala/cats/laws/SelectiveLaws.scala

```diff
@@ -0,0 +1,53 @@
+package cats
+package laws
+
+import cats.syntax.all._
+
+/**
+ * Laws that must be obeyed by any `Selective`.
+ */
+trait SelectiveLaws[F[_]] extends ApplicativeLaws[F] {
+  implicit override def F: Selective[F]
+
+  def selectiveIdentity[A, B](faa: F[Either[A, A]]): IsEq[F[A]] =
+    faa.select[A, A](F.pure(identity)) <-> faa.map(_.merge)
```

Comment:
this almost has to be true due to parametricity unless you just left one side or the other into an error, which I guess you could do for some Fs